protected/humhub/modules/admin/resources/js/humhub.admin.PendingRegistrations.js

```diff
@@ -0,0 +1,65 @@
+humhub.module('admin.PendingRegistrations', function(module, require, $) {
+    var object = require('util').object;
+    var Widget = require('ui.widget').Widget;
+    var client = require('client');
+    var urlDeleteSelected;
+
+    var PendingRegistrations = function(node, options) {
+        Widget.call(this, node, options);
+    };
+
+    object.inherits(PendingRegistrations, Widget);
+
+    PendingRegistrations.deleteAllSelected = function(evt) {
+        client.post(evt).then(function() {
+            var keys = $("#grid").yiiGridView("getSelectedRows");
+            $.ajax({
+                url: urlDeleteSelected,
+                type: "POST",
+                data: {id: keys},
+            })
+        }).catch(function(e) {
+            module.log.error(e, true);
+        })
+    };
+
+    PendingRegistrations.deleteAll = function(evt) {
+        client.post(evt).then(function() {
+            that.reload();
```

Comment:
That is currently not initialized. Add `var that = this` prior to the `client.post()` call.

Reply:
Done with client.reload()
Redirect in controller also works, but I did not want to leave empty  
client.post(evt).then(function()  like this:

```
        client.post(evt).then(function () {

        }).catch(function (e) {
            module.log.error(e, true);
        })
```


)

Reply:
This would have been an alternative:

```
client.post(evt)..catch(function (e) {
    module.log.error(e, true);
});
```

In this case you could just omit the `then` handler.

Reply:
Done!